source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,379 @@
+      Http::Headers::get().CacheControlValues.NoCache);
+  response_headers.insertConnection().value().setReference(
+      Http::Headers::get().ConnectionValues.Close);
+  response_headers.insertAccessControlAllowHeaders().value().setReference(
+      Http::Headers::get().AccessControlAllowHeadersValue.AccessControlAllowHeadersHystrix);
+  response_headers.insertAccessControlAllowOrigin().value().setReference(
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  Http::StreamDecoderFilterCallbacks& stream_decoder_filter_callbacks =
+      admin_stream.getDecoderFilterCallbacks();
+
+  registerConnection(&stream_decoder_filter_callbacks);
+
+  admin_stream.setEndStreamOnComplete(false); // set streaming
+
+  // Separated out just so it's easier to understand
+  auto on_destroy_callback = [this, &stream_decoder_filter_callbacks]() {
+    ENVOY_LOG(debug, "stopped sending data to hystrix dashboard on port {}",
+              stream_decoder_filter_callbacks.connection()->remoteAddress()->asString());
+
+    // Unregister the callbacks from the sink so data is no longer encoded through them.
+    unregisterConnection(&stream_decoder_filter_callbacks);
+  };
+
+  // Add the callback to the admin_filter list of callbacks
+  admin_stream.addOnDestroyCallback(std::move(on_destroy_callback));
+
+  ENVOY_LOG(debug, "started sending data to hystrix dashboard on port {}",
+            stream_decoder_filter_callbacks.connection()->remoteAddress()->asString());
+  return Http::Code::OK;
+}
+
+void HystrixSink::flush(Stats::Source&) {
+  if (callbacks_list_.empty())
+    return;
+  incCounter();
+  std::stringstream ss;
+  Upstream::ClusterManager::ClusterInfoMap clusters = server_.clusterManager().clusters();
+  for (auto& cluster : clusters) {
+    Upstream::ClusterInfoConstSharedPtr cluster_info = cluster.second.get().info();
+
+    // update rolling window with cluster stats
+    updateRollingWindowMap(cluster_info);
+
+    // append it to stream to be sent
+    addClusterStatsToStream(
+        cluster_info->name(),
+        cluster_info->resourceManager(Upstream::ResourcePriority::Default).pendingRequests().max(),
+        cluster_info->statsScope().gauge("membership_total").value(),
+        server_.statsFlushInterval().count(), ss);
+  }
+
+  Buffer::OwnedImpl data;
+  for (auto callbacks : callbacks_list_) {
+    data.add(ss.str());
+    callbacks->encodeData(data, false);
+  }
+
+  // send keep alive ping
+  // TODO (@trabetti) : is it ok to send together with data?
+  Buffer::OwnedImpl ping_data;
+  for (auto callbacks : callbacks_list_) {
+    ping_data.add(":\n\n");
+    callbacks->encodeData(ping_data, false);
+  }
+
+  // check if any clusters were removed, and remove from cache
+  if (clusters.size() < cluster_stats_cache_map_.size()) {
+    std::vector<std::string> clusters_to_remove;
+    for (auto& itr : cluster_stats_cache_map_) {
```

Comment:
nit: this could be made a bit simpler/faster by doing the following:
```
for (auto it = cluster_stats_cache_map_.begin(); it != cluster_stats_cache_map_.end();) {
  if (clusters.find(it.first) == clusters.end()) {
    it = cluster_stats_cache_map_.erase(it);
  } else {
    ++it;
  }
}